metadata-ws/src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/entities/Auditable.java

```diff
@@ -42,4 +43,8 @@ public void setLastModifiedDate(ZonedDateTime lastModifiedDate) {
     }
 
     public abstract ID getId();
+
+    public LocalDate getReleaseDate() {
```

Comment:
Wouldn't this be best as a pure virtual method, so all the `Auditable` classes are forced to implement it? The ones like `ReferenceSequence`, whose release date is not explicitly needed, would have to implement it just to return null; but on the other hand we wouldn't have bogus default implementations that return null when they shouldn't.

Reply:
I agree that this is better and more explicit in case of “always public” entities. Implemented `getReleaseDate()` as abstract method, changed `Taxonomy` and `ReferenceSequence` classes accordingly.